andela/bp-esa-backend
# Eliminate rate limit error when running test

#### What does this PR do?
Successfully mock/stub out all slackClient and sendMail function calls in tests so as to eliminate rate limit errors due to actual api calls

#### Description of Task to be completed?
- define custom `slack` object to envelop used slackClient methods, in other to successfully mock slack api calls in test
- replace slackClient method calls with method calls from custom `slack` object
- mock emailTransport.sendMail function in `execute jobs` and assert mock function was called

#### How should this be manually tested?
1. On terminal, run command test: `npm test`
2. Observe that slack rate limit warning no longer shows because slack/email api calls have been successfully mocked

#### Any background context you want to provide? 
Currently, slack/email rate limiting errors occur when running tests, this is because mocking of slack/email api calls were unsuccessful, thus allowing actual api calls within test. With this fix, slack/email api calls in tests would be successfully mocked/stubbed

#### What are the relevant pivotal tracker stories?
#168220512[https://www.pivotaltracker.com/story/show/168220512](url)
